src/fe/fe_base.C

```diff
@@ -744,25 +744,25 @@ void FEGenericBase<Real>::compute_dual_shape_coeffs ()
   dual_coeff.resize(sz, sz);
   DenseMatrix<Real> A(sz, sz), D(sz, sz);
 
-  const std::vector<Real> JxW = this->get_JxW();
-
-  for (auto i : index_range(phi))
-    for (auto qp : index_range(phi[i]))
+  const std::vector<Real> weights = qrule.get_weights();
```

Comment:
This should catch a reference, avoid making a copy of the weights.

Reply:
Oops I am sorry. Should have learned this from last time :)